### PR TITLE
[Exchanges] kucoin futures

### PR DESCRIPTION
**Pull request by GuillaumeDSM**
requires https://github.com/Drakkar-Software/OctoBot-Trading/pull/811

### CODE REVIEW

**Review by Herklos — COMMENTED** [review]
Great work!

File: Trading/Exchange/bybit/bybit_exchange.py [hunk 1]
```diff
@@ -232,6 +232,7 @@ def fix_ticker(self, raw, **kwargs):
     
     def parse_position(self, fixed, **kwargs):
         try:
+            # x contract value ?
```
[the remark this answers is not in the document]
Comment (GuillaumeDSM) [reply Trading/Exchange/bybit/bybit_exchange.py:235]: Hum we need to handle it properly, i'll add a todo

File: Trading/Exchange/kucoin/kucoin_exchange.py [hunk 2]
```diff
@@ -75,3 +106,249 @@ def should_log_on_ddos_exception(self, exception) -> bool:
...
         """
         return Kucoin.INSTANT_RETRY_ERROR_CODE not in str(exception)
+
+    def get_order_additional_params(self, order) -> dict:
+        params = {}
+        if self.exchange_manager.is_future:
+            contract = self.exchange_manager.exchange.get_pair_future_contract(order.symbol)
+            params["leverage"] = float(contract.current_leverage)
+            params["reduceOnly"] = order.reduce_only
+            params["closeOrder"] = order.close_position
```
Comment (Herklos) [inline Trading/Exchange/kucoin/kucoin_exchange.py:116]: 👍

**Review by Herklos — APPROVED** [review]
So many adaptations with Kucoin 😮 

Great job!

File: Trading/Exchange/kucoin/kucoin_exchange.py [hunk 3]
```diff
@@ -75,3 +106,249 @@ def should_log_on_ddos_exception(self, exception) -> bool:
...
+            else trading_enums.PositionMode.ONE_WAY
+        parsed_leverage = self.safe_decimal(
+            parsed, trading_enums.ExchangeConstantsPositionColumns.LEVERAGE.value, constants.ZERO
+        )
+        if parsed_leverage == constants.ZERO:
+            # on kucoin, fetched empty position don't have a leverage value. Since it's required within OctoBot,
+            # add it manually
+            symbol = parsed[trading_enums.ExchangeConstantsPositionColumns.SYMBOL.value]
+            if self.connector.exchange_manager.exchange.has_pair_future_contract(symbol):
+                parsed[trading_enums.ExchangeConstantsPositionColumns.LEVERAGE.value] = \
```
Comment (Herklos) [inline Trading/Exchange/kucoin/kucoin_exchange.py:349]: 👍

**Pull request merged**